### PR TITLE
feat: extend order list filter by buyer selection

### PR DESCRIPTION
**Pull request by SGrueber**
## PR Type

[x] Feature

## What Is the New Behavior?

The search filter of the order list has been extended. The account admin of a b2b customer can now see the orders of all buyers of his organization and also filter them by a certain buyer. 

## Does this PR Introduce a Breaking Change?

[x] No

## Other Information

This functionality is only available with the ICM 12.1.0 and higher.

[AB#98149](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/98149)

### CODE REVIEW

**Comment by iwiederhold** [comment]
Please check the distance between the magnifying glass and the labelling of the FIND ORDER button. This is a cheat. :-) 

image

File: src/assets/i18n/en_US.json [hunk 1]
```diff
@@ -219,9 +219,11 @@
   "account.order.questions.title": "Questions?",
   "account.order.subtitle": "The most recent order appears first. Please allow up to 5 minutes for new orders to appear below.",
   "account.order.view_all_order.link": "View all orders",
+  "account.order_history.filter.all_buyers": "All Buyers",
```
Comment (M-Behr) [inline src/assets/i18n/en_US.json:222]: We use sentence case now: "All Buyers" --> "All buyers"

**Review by M-Behr — APPROVED** [review]
Good work! :)

**Review by iwiederhold — APPROVED** [review]

**Review by skoch-intershop — APPROVED** [review]

**Comment by iwiederhold** [comment]
The validation is not necessary here.

image

**Review by M-Behr — APPROVED** [review]
Looks good!

**Review by iwiederhold — APPROVED** [review]

**Review by shauke — APPROVED** [review]

**Pull request merged**